### DISCUSSION TITLE
Trigger update for Docker image using Azure Container Registry cache feature

### DISCUSSION THREAD

**Discussion by marcmognol** [renovatebot/renovate#25491] [Suggest an Idea]
### Tell us more.

# Feature request

I would like to know if it could possible that Renovate trigger PR for Docker image hosted in Docker Hub but pulled through Azure Container Registry cache feature.

You can learn more about cache feature for ACR here:
[https://learn.microsoft.com/en-us/azure/container-registry/tutorial-artifact-cache](url)

Please note that ACR will cache the image only when it is pulled the **first time**.

## My use case

Today, I build my Dockerfile like this, and Renovate suggests PR to update digest or new version of this image:
```
FROM mcr.microsoft.com/dotnet/runtime:7.0.12-alpine3.18@sha256:d6b7573ce1c0a5ede5385855a76d7d1e6ece9bb10fc6140d0858a7aac895b51e AS base
WORKDIR /app
EXPOSE 8080
…
```

Tomorrow, I would like to use Azure ACR cache feature to have a “copy” of these images in our registry, to do so I’ll convert my Dockerfile like the following:

```
FROM myownregistry.azureacr.io/dotnet-runtime:7.0.12-alpine3.18@sha256:d6b7573ce1c0a5ede5385855a76d7d1e6ece9bb10fc6140d0858a7aac895b51e AS base
WORKDIR /app
EXPOSE 8080
…
```

**Notice** that the **registry and repository** are not the same !

## Question
- Is it possible to reproduce this behavior with regex rules ?
- Could you please guide me to achieve this goal ?

Thank you :smile:

**Comment by viceice** [comment]
it should work as long as the cache returns the full tags list and not only the cached tags.

if it doesn't work then try the renovate registry aliases config option.

**Reply by marcmognol** [reply]
Hi @viceice,
Thank you for your quick reply.
I think registryAliases is the solution.

Thanks again.
Marc
